org.faktorips.aggregator/targets/pom.xml

```diff
@@ -12,7 +12,6 @@
         <relativePath>../pom.xml</relativePath>
     </parent>
 
-    <groupId>org.faktorips</groupId>
```

Comment:
Warum ist dieses POM noch hier, wenn alle darin aufgezählten Module gelöscht wurden?

Reply:
Die Module sind nicht gelöscht sonder nur das pom.xml (diese wird automatisch von Tycho generiert während des builds).

Reply:
Ok, wie muss ich dann den Build aufrufen? Bei Maven Clean Install auf dem base pom.xml meckert Maven:
`[ERROR]   The project org.faktorips:org.faktorips.aggregator.targets:22.6.0-SNAPSHOT (/git/faktorips.base/org.faktorips.aggregator/targets/pom.xml) has 12 errors
[ERROR]     Child module /git/faktorips.base/org.faktorips.aggregator/targets/2019-03/pom.xml of /git/faktorips.base/org.faktorips.aggregator/targets/pom.xml does not exist
[ERROR]     Child module /git/faktorips.base/org.faktorips.aggregator/targets/2019-06/pom.xml of /git/faktorips.base/org.faktorips.aggregator/targets/pom.xml does not exist
[ERROR]     Child module /git/faktorips.base/org.faktorips.aggregator/targets/2019-09/pom.xml of /git/faktorips.base/org.faktorips.aggregator/targets/pom.xml does not exist
[ERROR]     Child module /git/faktorips.base/org.faktorips.aggregator/targets/2019-12/pom.xml of /git/faktorips.base/org.faktorips.aggregator/targets/pom.xml does not exist
[ERROR]     Child module /git/faktorips.base/org.faktorips.aggregator/targets/2020-03/pom.xml of /git/faktorips.base/org.faktorips.aggregator/targets/pom.xml does not exist
[ERROR]     Child module /git/faktorips.base/org.faktorips.aggregator/targets/2020-06/pom.xml of /git/faktorips.base/org.faktorips.aggregator/targets/pom.xml does not exist
[ERROR]     Child module /git/faktorips.base/org.faktorips.aggregator/targets/2020-09/pom.xml of /git/faktorips.base/org.faktorips.aggregator/targets/pom.xml does not exist
[ERROR]     Child module /git/faktorips.base/org.faktorips.aggregator/targets/2020-12/pom.xml of /git/faktorips.base/org.faktorips.aggregator/targets/pom.xml does not exist
[ERROR]     Child module /git/faktorips.base/org.faktorips.aggregator/targets/2021-03/pom.xml of /git/faktorips.base/org.faktorips.aggregator/targets/pom.xml does not exist
[ERROR]     Child module /git/faktorips.base/org.faktorips.aggregator/targets/2021-06/pom.xml of /git/faktorips.base/org.faktorips.aggregator/targets/pom.xml does not exist
[ERROR]     Child module /git/faktorips.base/org.faktorips.aggregator/targets/2021-09/pom.xml of /git/faktorips.base/org.faktorips.aggregator/targets/pom.xml does not exist
[ERROR]     Child module /git/faktorips.base/org.faktorips.aggregator/targets/2021-12/pom.xml of /git/faktorips.base/org.faktorips.aggregator/targets/pom.xml does not exist
`

Reply:
Klappt bei mir leider nicht. Ich bin nicht sicher, ob ich evtl. in meiner settings.xml noch was umkonfigurieren muss, da die build extension: Plugin org.faktorips:codequality-config:1.0.0-SNAPSHOT nicht gefunden wird, selbst wenn ich sie vorher explizit baue...

`(base) schwering /git/faktorips.base (issue_13)$ gh pr checkout 16 --force
From github.com:faktorips/faktorips.base
 * branch                  refs/pull/16/head -> FETCH_HEAD
HEAD is now at 6eeadafdf7 Integrate code-styles and enable two more modules
(base) schwering /git/faktorips.base (issue_13)$ mvn clean install -f org.faktorips.codequality.config/
[WARNING] The POM for org.eclipse.tycho:tycho-build:jar:2.7.0-SNAPSHOT is missing, no dependency information available
[WARNING] Failed to read extensions descriptor /git/faktorips.base/.mvn/extensions.xml: Plugin org.eclipse.tycho:tycho-build:2.7.0-SNAPSHOT or one of its dependencies could not be resolved: Failure to find org.eclipse.tycho:tycho-build:jar:2.7.0-SNAPSHOT in https://nexus.faktorzehn.de/content/groups/private was cached in the local repository, resolution will not be reattempted until the update interval of nexus has elapsed or updates are forced
[INFO] Scanning for projects...
[INFO] 
[INFO] ------------------< org.faktorips:codequality-config >------------------
[INFO] Building Faktor IPS Build Configurations 1.0.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ codequality-config ---
[INFO] Deleting /git/faktorips.base/org.faktorips.codequality.config/target
[INFO] 
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ codequality-config ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 3 resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.2:compile (default-compile) @ codequality-config ---
[INFO] No sources to compile
[INFO] 
[INFO] --- maven-resources-plugin:2.6:testResources (default-testResources) @ codequality-config ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /git/faktorips.base/org.faktorips.codequality.config/src/test/resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.2:testCompile (default-testCompile) @ codequality-config ---
[INFO] No sources to compile
[INFO] 
[INFO] --- maven-surefire-plugin:2.22.0:test (default-test) @ codequality-config ---
[INFO] No tests to run.
[INFO] 
[INFO] --- maven-jar-plugin:2.6:jar (default-jar) @ codequality-config ---
[INFO] Building jar: /git/faktorips.base/org.faktorips.codequality.config/target/codequality-config-1.0.0-SNAPSHOT.jar
[INFO] 
[INFO] --- maven-install-plugin:2.4:install (default-install) @ codequality-config ---
[INFO] Installing /git/faktorips.base/org.faktorips.codequality.config/target/codequality-config-1.0.0-SNAPSHOT.jar to /home/schwering/.m2/repository/org/faktorips/codequality-config/1.0.0-SNAPSHOT/codequality-config-1.0.0-SNAPSHOT.jar
[INFO] Installing /git/faktorips.base/org.faktorips.codequality.config/pom.xml to /home/schwering/.m2/repository/org/faktorips/codequality-config/1.0.0-SNAPSHOT/codequality-config-1.0.0-SNAPSHOT.pom
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.680 s
[INFO] Finished at: 2022-02-07T11:02:58+01:00
[INFO] ------------------------------------------------------------------------
(base) schwering /git/faktorips.base (issue_13)$ mvn clean install -V
Apache Maven 3.6.3
Maven home: /usr/share/maven
Java version: 11.0.14, vendor: Eclipse Adoptium, runtime: /home/schwering/.sdkman/candidates/java/11.0.14-tem
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "5.13.0-28-generic", arch: "amd64", family: "unix"
[WARNING] The POM for org.eclipse.tycho:tycho-build:jar:2.7.0-SNAPSHOT is missing, no dependency information available
[WARNING] Failed to read extensions descriptor /git/faktorips.base/.mvn/extensions.xml: Plugin org.eclipse.tycho:tycho-build:2.7.0-SNAPSHOT or one of its dependencies could not be resolved: Failure to find org.eclipse.tycho:tycho-build:jar:2.7.0-SNAPSHOT in https://nexus.faktorzehn.de/content/groups/private was cached in the local repository, resolution will not be reattempted until the update interval of nexus has elapsed or updates are forced
[INFO] Scanning for projects...
[ERROR] [ERROR] Some problems were encountered while processing the POMs:
[ERROR] Child module /git/faktorips.base/org.faktorips.aggregator/targets/2019-03/pom.xml of /git/faktorips.base/org.faktorips.aggregator/targets/pom.xml does not exist @ 
[ERROR] Child module /git/faktorips.base/org.faktorips.aggregator/targets/2019-06/pom.xml of /git/faktorips.base/org.faktorips.aggregator/targets/pom.xml does not exist @ 
[ERROR] Child module /git/faktorips.base/org.faktorips.aggregator/targets/2019-09/pom.xml of /git/faktorips.base/org.faktorips.aggregator/targets/pom.xml does not exist @ 
[ERROR] Child module /git/faktorips.base/org.faktorips.aggregator/targets/2019-12/pom.xml of /git/faktorips.base/org.faktorips.aggregator/targets/pom.xml does not exist @ 
[ERROR] Child module /git/faktorips.base/org.faktorips.aggregator/targets/2020-03/pom.xml of /git/faktorips.base/org.faktorips.aggregator/targets/pom.xml does not exist @ 
[ERROR] Child module /git/faktorips.base/org.faktorips.aggregator/targets/2020-06/pom.xml of /git/faktorips.base/org.faktorips.aggregator/targets/pom.xml does not exist @ 
[ERROR] Child module /git/faktorips.base/org.faktorips.aggregator/targets/2020-09/pom.xml of /git/faktorips.base/org.faktorips.aggregator/targets/pom.xml does not exist @ 
[ERROR] Child module /git/faktorips.base/org.faktorips.aggregator/targets/2020-12/pom.xml of /git/faktorips.base/org.faktorips.aggregator/targets/pom.xml does not exist @ 
[ERROR] Child module /git/faktorips.base/org.faktorips.aggregator/targets/2021-03/pom.xml of /git/faktorips.base/org.faktorips.aggregator/targets/pom.xml does not exist @ 
[ERROR] Child module /git/faktorips.base/org.faktorips.aggregator/targets/2021-06/pom.xml of /git/faktorips.base/org.faktorips.aggregator/targets/pom.xml does not exist @ 
[ERROR] Child module /git/faktorips.base/org.faktorips.aggregator/targets/2021-09/pom.xml of /git/faktorips.base/org.faktorips.aggregator/targets/pom.xml does not exist @ 
[ERROR] Child module /git/faktorips.base/org.faktorips.aggregator/targets/2021-12/pom.xml of /git/faktorips.base/org.faktorips.aggregator/targets/pom.xml does not exist @ 
[ERROR] Unresolveable build extension: Plugin org.faktorips:codequality-config:1.0.0-SNAPSHOT or one of its dependencies could not be resolved: Failed to read artifact descriptor for org.faktorips:codequality-config:jar:1.0.0-SNAPSHOT @ 
 @ 
[ERROR] The build could not read 2 projects -> [Help 1]
[ERROR]   
[ERROR]   The project org.faktorips:org.faktorips.aggregator.targets:22.6.0-SNAPSHOT (/git/faktorips.base/org.faktorips.aggregator/targets/pom.xml) has 12 errors
[ERROR]     Child module /git/faktorips.base/org.faktorips.aggregator/targets/2019-03/pom.xml of /git/faktorips.base/org.faktorips.aggregator/targets/pom.xml does not exist
[ERROR]     Child module /git/faktorips.base/org.faktorips.aggregator/targets/2019-06/pom.xml of /git/faktorips.base/org.faktorips.aggregator/targets/pom.xml does not exist
[ERROR]     Child module /git/faktorips.base/org.faktorips.aggregator/targets/2019-09/pom.xml of /git/faktorips.base/org.faktorips.aggregator/targets/pom.xml does not exist
[ERROR]     Child module /git/faktorips.base/org.faktorips.aggregator/targets/2019-12/pom.xml of /git/faktorips.base/org.faktorips.aggregator/targets/pom.xml does not exist
[ERROR]     Child module /git/faktorips.base/org.faktorips.aggregator/targets/2020-03/pom.xml of /git/faktorips.base/org.faktorips.aggregator/targets/pom.xml does not exist
[ERROR]     Child module /git/faktorips.base/org.faktorips.aggregator/targets/2020-06/pom.xml of /git/faktorips.base/org.faktorips.aggregator/targets/pom.xml does not exist
[ERROR]     Child module /git/faktorips.base/org.faktorips.aggregator/targets/2020-09/pom.xml of /git/faktorips.base/org.faktorips.aggregator/targets/pom.xml does not exist
[ERROR]     Child module /git/faktorips.base/org.faktorips.aggregator/targets/2020-12/pom.xml of /git/faktorips.base/org.faktorips.aggregator/targets/pom.xml does not exist
[ERROR]     Child module /git/faktorips.base/org.faktorips.aggregator/targets/2021-03/pom.xml of /git/faktorips.base/org.faktorips.aggregator/targets/pom.xml does not exist
[ERROR]     Child module /git/faktorips.base/org.faktorips.aggregator/targets/2021-06/pom.xml of /git/faktorips.base/org.faktorips.aggregator/targets/pom.xml does not exist
[ERROR]     Child module /git/faktorips.base/org.faktorips.aggregator/targets/2021-09/pom.xml of /git/faktorips.base/org.faktorips.aggregator/targets/pom.xml does not exist
[ERROR]     Child module /git/faktorips.base/org.faktorips.aggregator/targets/2021-12/pom.xml of /git/faktorips.base/org.faktorips.aggregator/targets/pom.xml does not exist
[ERROR]   
[ERROR]   The project org.faktorips:faktorips-maven-plugin:22.6.0-SNAPSHOT (/git/faktorips.base/faktorips-maven-plugin/pom.xml) has 1 error
[ERROR]     Unresolveable build extension: Plugin org.faktorips:codequality-config:1.0.0-SNAPSHOT or one of its dependencies could not be resolved: Failed to read artifact descriptor for org.faktorips:codequality-config:jar:1.0.0-SNAPSHOT: Failure to find org.faktorips:base:pom:1.0.0-SNAPSHOT in https://nexus.faktorzehn.de/content/groups/private was cached in the local repository, resolution will not be reattempted until the update interval of nexus has elapsed or updates are forced -> [Help 2]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/ProjectBuildingException
[ERROR] [Help 2] http://cwiki.apache.org/confluence/display/MAVEN/PluginManagerException
`

Reply:
Versuch mal wie hier beschrieben unter "If you using the tycho-pomless extension or you want to make the Tycho development version available in all your builds" das Snapshot repo in die `settingx.xml `einzufügen:
https://github.com/eclipse/tycho/wiki#getting-tycho-snapshots

Ich glaube die Buildextension wird bei dir nicht gefunden.

Reply:
Wenn ich mir den Log genauer anschaue, liegt's wohl daran, dass ich  tycho-snapshots nicht in der settings.xml hab

Reply:
Funktioniert es wenn du die extension vorher per mvn install installierst? Ich bin mir nicht ganz sicher ob extensions teil des Reactor-Builds sein können.

Reply:
Leider nein:
`(base) schwering /git/faktorips.base (issue_13)$ mvn clean install -f org.faktorips.codequality.config/
[INFO] Scanning for projects...
[INFO] 
[INFO] ------------------< org.faktorips:codequality-config >------------------
[INFO] Building Faktor IPS Build Configurations 1.0.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ codequality-config ---
[INFO] Deleting /git/faktorips.base/org.faktorips.codequality.config/target
[INFO] 
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ codequality-config ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 3 resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.2:compile (default-compile) @ codequality-config ---
[INFO] No sources to compile
[INFO] 
[INFO] --- maven-resources-plugin:2.6:testResources (default-testResources) @ codequality-config ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /git/faktorips.base/org.faktorips.codequality.config/src/test/resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.2:testCompile (default-testCompile) @ codequality-config ---
[INFO] No sources to compile
[INFO] 
[INFO] --- maven-surefire-plugin:2.22.0:test (default-test) @ codequality-config ---
[INFO] No tests to run.
[INFO] 
[INFO] --- maven-jar-plugin:2.6:jar (default-jar) @ codequality-config ---
[INFO] Building jar: /git/faktorips.base/org.faktorips.codequality.config/target/codequality-config-1.0.0-SNAPSHOT.jar
[INFO] 
[INFO] --- maven-install-plugin:2.4:install (default-install) @ codequality-config ---
[INFO] Installing /git/faktorips.base/org.faktorips.codequality.config/target/codequality-config-1.0.0-SNAPSHOT.jar to /home/schwering/.m2/repository/org/faktorips/codequality-config/1.0.0-SNAPSHOT/codequality-config-1.0.0-SNAPSHOT.jar
[INFO] Installing /git/faktorips.base/org.faktorips.codequality.config/pom.xml to /home/schwering/.m2/repository/org/faktorips/codequality-config/1.0.0-SNAPSHOT/codequality-config-1.0.0-SNAPSHOT.pom
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.652 s
[INFO] Finished at: 2022-02-07T11:33:33+01:00
[INFO] ------------------------------------------------------------------------
(base) schwering /git/faktorips.base (issue_13)$ mvn clean install -V
Apache Maven 3.8.4 (9b656c72d54e5bacbed989b64718c159fe39b537)
Maven home: /home/schwering/opt/apache-maven-3.8.4
Java version: 11.0.14, vendor: Eclipse Adoptium, runtime: /home/schwering/.sdkman/candidates/java/11.0.14-tem
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "5.13.0-28-generic", arch: "amd64", family: "unix"
[INFO] Scanning for projects...
[ERROR] [ERROR] Some problems were encountered while processing the POMs:
[ERROR] Unresolveable build extension: Plugin org.faktorips:codequality-config:1.0.0-SNAPSHOT or one of its dependencies could not be resolved: Failed to read artifact descriptor for org.faktorips:codequality-config:jar:1.0.0-SNAPSHOT @ 
 @ 
[ERROR] The build could not read 1 project -> [Help 1]
[ERROR]   
[ERROR]   The project org.faktorips:faktorips-maven-plugin:22.6.0-SNAPSHOT (/git/faktorips.base/faktorips-maven-plugin/pom.xml) has 1 error
[ERROR]     Unresolveable build extension: Plugin org.faktorips:codequality-config:1.0.0-SNAPSHOT or one of its dependencies could not be resolved: Failed to read artifact descriptor for org.faktorips:codequality-config:jar:1.0.0-SNAPSHOT: org.faktorips:base:pom:1.0.0-SNAPSHOT was not found in https://repo.eclipse.org/content/repositories/tycho-snapshots/ during a previous attempt. This failure was cached in the local repository and resolution is not reattempted until the update interval of tycho-snapshots has elapsed or updates are forced -> [Help 2]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/ProjectBuildingException
[ERROR] [Help 2] http://cwiki.apache.org/confluence/display/MAVEN/PluginManagerException
`

Reply:
> This failure was cached in the local repository and resolution is not reattempted until the update interval of tycho-snapshots has elapsed or updates are forced

Kanst du ex mal mit der -U option probieren:

```
-U,--update-snapshots                  Forces a check for missing
                                        releases and updated snapshots on
                                        remote repositories
```

Reply:
Ändert leider nichts.

Reply:
Hm... vieleicht einmal mit ganz leerem maven repository?

`faktorips.base$`  `mvn clean install -DskipTests -DlocalRepository=/tmp/testm2`

läuft bei mir ohne weitere Aktionen durch.